src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
+
+  takeDamage(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide(hidden = true) {
+    this.hidden = hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey.hidden === false) {
+      prey.takeDamage(50);
+    }
```

Comment:
The `bite` method should include a check to ensure that the `prey` is an instance of the `Herbivore` class before attempting to bite. This can be done using `if (prey instanceof Herbivore && !prey.hidden)`. This is necessary to meet the task requirement of only allowing carnivores to bite herbivores.